### PR TITLE
feat: replace saveBills by saveBankingDocuments

### PR DESCRIPTION
**Pull request by doubleface**
This PR deprecates saveBills and replace is with saveDocuments to allow it to handle more doctypes
which may need saved and linked to bank operations.

The doctype stays `io.cozy.bills` by default and an option has been added to specify the doctype :

```js
saveDocuments(documents, fields, {
  doctype: 'io.cozy.payslips'
})
```

At the moment :

- io.cozy.bills
- io.cozy.payslips

These doctype have some attributes in common :

- amount
- date
- vendor
- $relationships.data.document

These attributes will make it easier (hopefully) for the cozy-bank application to link theses
documents to bank operations.

### CODE REVIEW

**Comment by ptbrowne** [comment]
It is not clear with `saveDocuments` that it will handle banking specific doctypes. Maybe there could be specific default `postProcess` actions that are taken for specific doctypes ?

**Comment by aenario** [comment]
@ptbrowne : Trying to clarify your comment, do you want 

**A.** a more specific name for this function `saveBankingDocuments` 
**B.** a smart `saveDocuments` with a switch for all doctypes : 
```
module.exports = (entries, fields, options = {}) => {
  switch(options.doctype) {
     case 'io.cozy.bills':
     case 'io.cozy.payslips':
        linkBankOperations(...)
     default:
         noop()
  }
```

**Comment by ptbrowne** [comment]
The B. To me, there should be only 1 function to saveData and then specific plugins for each doctypes. I think it is better this way since it will be less complex from the konnector developer point of view (only 1 function to know) and more flexible from the library point of view (less API surface and more control over which plugin works on the data).

```
registerPlugin('io.cozy.bills', linkBankOperations)
registerPlugin('io.cozy.payslips', linkBankOperations) 
```

Anyway, in this specific case, the linking will be extracted as a Bank service so I am not sure we should dwell on this issue for too long ;)

**Comment by doubleface** [comment]
Here is the renaming and the associated generated documentation.

After this change of name, I rename also the `banking` option to `noBanking` and removed the list of doctypes with banking.

It is now just possible to deactivate the linkBankOperation. It will be enough for our use at the moment and we will not need to change cozy-konnector-libs for each new doctype.

**Comment by aenario** [comment]
:+1: Good enough for now, let's merge !

File: packages/cozy-konnector-libs/src/libs/saveBankingDocuments.js [hunk 1]
```diff
@@ -41,20 +45,32 @@ module.exports = (entries, fields, options = {}) => {
...
 
   const originalEntries = entries
   return saveFiles(entries, fields, options)
-    .then(entries => hydrateAndFilter(entries, DOCTYPE, options))
-    .then(entries => addData(entries, DOCTYPE, options))
-    .then(entries => linkBankOperations(originalEntries, DOCTYPE, fields, options))
+    .then(entries => hydrateAndFilter(entries, options.doctype, options))
+    .then(entries => addData(entries, options.doctype, options))
+    .then(entries => {
+      if (!options.banking) return entries
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply packages/cozy-konnector-libs/src/libs/saveBankingDocuments.js]: IMHO I should this very weird to have boolean option that are negative.

http://truelogic.org/wordpress/2009/02/27/how-not-to-name-a-boolean-variable/

**Pull request merged**